src/makielayout/blocks/legend.jl

```diff
@@ -794,6 +945,44 @@ function legend_position_to_aligns(t::Tuple{Any, Any})
     (halign = t[1], valign = t[2])
 end
 
+function toggle_visibility!(entry::LegendEntry, sync = false)
+    for el in entry.elements
+        if (el !== nothing) && hasfield(typeof(el), :plots) && (el.plots !== nothing)
+            for plot in el.plots
+                !hasproperty(plot, :visible) && continue
+                plot.visible[] = sync ? true : !plot.visible[]
+            end
+        end
+    end
+    return
+end
+
+function get_plot_visibilities(entry::LegendEntry)
+    visibilities = Observable{Bool}[]
+    for element in entry.elements
+        if (element !== nothing) && hasfield(typeof(element), :plots) && (element.plots !== nothing)
+            for plot in element.plots
+                !hasproperty(plot, :visible) && continue
```

Comment:
I feel like we should just guarantee that every plot has a visibility field.

Reply:
We should, but I see that as wasted effort before #4630.